app/tutorials/traffic-light-communication/src/main/java/org/eclipse/mosaic/app/tutorial/TrafficLightApp.java

```diff
@@ -65,13 +65,17 @@ private void setRed() {
     public void onMessageReceived(ReceivedV2xMessage receivedV2xMessage) {
         if (receivedV2xMessage.getMessage() instanceof GreenWaveMsg) {
             getLog().infoSimTime(this, "Received GreenWaveMsg");
-            if (((GreenWaveMsg) receivedV2xMessage.getMessage()).getMessage().equals(SECRET)) {
-                getLog().infoSimTime(this, "Received correct passphrase: {}", SECRET);
-
-                if (DEFAULT_PROGRAM.equals(getOs().getCurrentProgram().getProgramId())) {
-                    setGreen();
+            if (receivedV2xMessage.getMessage().getRouting().getSource().getSourcePosition()
+                    .distanceTo(getOs().getPosition()) <= 15) {
```

Comment:
Magic numbers somewhere in the code are not preferable. Instead, create a constant with a meaningful name (e.g. `MIN_DISTANCE`) and use it here.